### PR TITLE
Add usage of a study-specific representation of the data table

### PR DESCRIPTION
**Pull request by ewelinagr**
To be merged after the back-end part: https://github.com/thehyve/transmart-core/pull/218

TMT-154

### CODE REVIEW

**Comment by ewelinagr** [comment]
@beaugogh I tested it, with fixes from the last commit and from [PR#349](https://github.com/thehyve/glowing-bear/pull/349) it is working fine

File: src/app/models/transmart-models/transmart-default-table-representation.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export class TransmartDefaultTableRepresentation {
```
Comment (beaugogh) [inline src/app/models/transmart-models/transmart-default-table-representation.ts]: isn't a bit redundant now that we have TransmartTableState and TransmartDefaultTableRepresentation? TransmartDefaultTableRepresentation is just a special case of TransmartTableState, right?

File: src/app/models/transmart-models/transmart-studies-dimensions.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {Dimension} from "../table-models/dimension";
+import {TransmartTableState} from "app/models/transmart-models/transmart-table-state";
+
+export class TransmartStudiesDimensions {
+  availableDimensions: Array<Dimension> = [];
+  defaultTableRepresentation: TransmartTableState;
```
Comment (beaugogh) [inline src/app/models/transmart-models/transmart-studies-dimensions.ts]: 'defaultTableRepresentation: TransmartTableState' is confusing when compared to 'defaultTabularRepresentation: TransmartDefaultTableRepresentation'

File: src/app/services/resource.service.ts [hunk 3]
```diff
@@ -279,51 +280,30 @@ export class ResourceService {
...
-
-          commonDimensions.forEach((name: string) => {
-            if (highDims.indexOf(name) === -1) {
-              dimensions.push(new Dimension(name));
-            }
-          });
-        }
-
-        return dimensions;
+        return TransmartMapper.mapStudiesDimensions(transmartStudies);
```
Comment (beaugogh) [inline src/app/services/resource.service.ts]: also here? studiesDimensions

File: src/app/services/transmart-resource/transmart-mapper.ts [hunk 4]
```diff
@@ -236,6 +238,103 @@ export class TransmartMapper {
...
+          dimensions.push(new Dimension(name));
+        }
+      });
+      dimensions.forEach((dimension: Dimension) => studiesDimensions.availableDimensions.push(dimension));
+    }
+
+    return studiesDimensions;
+  }
+
+  public static mapDefaultDimensionsRepresentation(studiesDimensions: TransmartStudyDimensions, dataTable: DataTable) {
```
Comment (beaugogh) [inline src/app/services/transmart-resource/transmart-mapper.ts]: normally if we create a mapper function, it maps something to something else, so maybe it should just take in the transmart study dims, and output the row/col dims that data table can use?

File: src/app/services/transmart-resource/transmart-mapper.ts [hunk 5]
```diff
@@ -236,6 +238,103 @@ export class TransmartMapper {
     return elements;
   }
 
+  public static mapStudiesDimensions(transmartStudies: TransmartStudy[]) {
```
Comment (beaugogh) [inline src/app/services/transmart-resource/transmart-mapper.ts]: specify the output class of this function

File: src/app/models/transmart-models/transmart-study-metadata.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import {TransmartTableState} from "./transmart-table-state";
+
+export class TransmartStudyMetadata {
+  defaultTabularRepresentation: TransmartTableState;
```
Comment (beaugogh) [inline src/app/models/transmart-models/transmart-study-metadata.ts]: 'default tabular representation '? maybe there is no such thing as default?

File: src/app/models/transmart-models/transmart-study-dimensions.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import {Dimension} from "../table-models/dimension";
+import {TransmartTableState} from "app/models/transmart-models/transmart-table-state";
+
+export class TransmartStudyDimensions {
+  availableDimensions: Array<Dimension> = [];
+  defaultTableRepresentation: TransmartTableState;
```
Comment (beaugogh) [inline src/app/models/transmart-models/transmart-study-dimensions.ts]: 'default table representation'

**Comment by beaugogh** [comment]
looks like we need to rebase

**Comment by ewelinagr** [comment]
@beaugogh The back-end PR is merged now, so this could be safely merged as well now.

**Pull request merged**